src/MassTransit.Abstractions/Internals/Extensions/TaskExtensions.cs

```diff
@@ -105,17 +105,32 @@ async Task<T> WaitAsync()
 
             async Task WaitAsync()
```

Comment:
The `cancellationToken` should be passed as argument to the local function.
Avoids the compiler generated display-type.

Reply:
it doesn't make any difference because as long as you do not make this method `static`. We still need to pass task + cancellation token